index.bs

```diff
@@ -1321,6 +1598,123 @@ urlPrefix: https://w3c.github.io/FileAPI/#dfn-; type: dfn;
 
 	</div><!-- algorithm -->
 
+	<div class="algorithm" data-algorithm="write-blobs-to-clipboard">
+	<h3 id="to-write-blobs-to-clipboard"><dfn>write blobs and option to the clipboard</dfn></h3>
+
+		: Input
+		:: |items|, a [=sequence=]&lt;{{Blob}}&gt;
+		:: |presentationStyle|, a [=ClipboardItem/clipboard item=]'s [=presentation style=]
+
+		: Output
+		:: None
+
+		1. For each |item| in |items|:
+
+			1. Let |formatString| be the result of running [=os specific well-known format=] given |item|'s {{Blob/type}}.
+
+			1. If |formatString| is empty then abort these steps.
+
+			1. Let |payload| be the result of [=UTF-8 decoding=] |item|'s underlying byte sequence.
+
+			1. Insert |payload| and |presentationStyle| into the [=system clipboard=] using |formatString| as the native clipboard format.
+
+	</div><!-- algorithm -->
+
+	<div class="algorithm" data-algorithm="os-specific-well-known-format">
+	<h3 id="to-os-specific-well-known-format"><dfn>os specific well-known format</dfn></h3>
+
+		: Input
+		:: |mimeType|, a {{Blob/type}}
+
+		: Output
+		:: |wellKnownFormat|, a platform specific string type. On Mac it's NSString, on Windows and Linux it is fetched from the atom table.
+
+		<p class=note>
+		For Windows see https://docs.microsoft.com/en-us/windows/win32/dataxchg/standard-clipboard-formats and
+		https://docs.microsoft.com/en-us/windows/win32/dataxchg/about-atom-tables?redirectedfrom=MSDN
+		For Mac see https://developer.apple.com/documentation/appkit/nspasteboardtype
+		</p>
+
+		1. Let |wellKnownFormat| be a string.
+
+		1. If |mimeType|'s [=MIME type/essence=] is "text/plain", then
+
+				On Windows, follow the convention described below:
+
+				1. Assign CF_UNICODETEXT to |wellKnownFormat|.
+
+				1. Return |wellKnownFormat|.
+
+				On MacOS, follow the convention described below:
+
+				1. Assign NSPasteboardTypeString to |wellKnownFormat|.
+
+				1. Return |wellKnownFormat|.
+
+				On Linux, ChromeOS, and Android, follow the convention described below:
+
+				1. Assign "text/plain" to |wellKnownFormat|.
+
+				1. Return |wellKnownFormat|.
+
+		1. Else, if |mimeType|'s [=MIME type/essence=] is "text/html", then
+
+				On Windows, follow the convention described below:
+
+				1. Assign CF_HTML to |wellKnownFormat|.
+
+				1. Return |wellKnownFormat|.
+
+				On MacOS, follow the convention described below:
+
+				1. Assign NSHTMLPboardType to |wellKnownFormat|.
+
+				1. Return |wellKnownFormat|.
+
+				On Linux, ChromeOS, and Android, follow the convention described below:
+
+				1. Assign "text/html" to |wellKnownFormat|.
+
+				1. Return |wellKnownFormat|.
+
+		1. Else, if |mimeType|'s [=MIME type/essence=] is "image/png", then
+
+				On Windows, follow the convention described below:
+
+				1. Assign "PNG" to |wellKnownFormat|.
```

Comment:
Is "PNG" the correct type? It's not included in https://docs.microsoft.com/en-us/windows/win32/dataxchg/standard-clipboard-formats#constants.

Reply:
In Chromium we use either "PNG" or `CF_DIBV5`. Not sure about other browsers.